### ISSUE TITLE
decrease ixn cost of grooming - LAYOUT AND BEHAVIOR changes

### ISSUE THREAD

**Issue by justinscherer** [ushahidi/platform#1896]
As a manager, I want the platform to help me perform actions to posts in the most efficient way possible because my time is valuable and situations may be urgent.

**Comment by justinscherer** [comment]
Hi all, I have finished the screenflows for the behaviour of posts in desktop. Please have a look at the PDF attached. 

More screens coming this week. 

[layout and behaviour screen flow desktop.pdf](https://github.com/ushahidi/platform/files/1332884/layout.and.behaviour.screen.flow.desktop.pdf)

**Comment by aoduor** [comment]
Questions on unlocking
1. Is unlocking limited to just admins, or can ANYONE unlock?
2. I can see we send a notification to someone viewing a post about it being unlocked, do we do the same for someone who's editing a post and being "Kicked off"? Or is this the same thing?

**Comment by aoduor** [comment]
On filtering and sorting
1. We're missing the ability to sign up for notifications and limit audience of saved searches during creation.. Or is that on purpose too?

**Comment by aoduor** [comment]
<3 Changelog + Layout... all good on my end.

**Comment by justinscherer** [comment]
> Is unlocking limited to just admins, or can ANYONE unlock?

Only admins


> I can see we send a notification to someone viewing a post about it being unlocked, do we do the same for someone who's editing a post and being "Kicked off"? Or is this the same thing?

Same thing

**Comment by justinscherer** [comment]
On filtering and sorting

> We're missing the ability to sign up for notifications, limit audience of saved searches during creation.. Or is that on purpose too?

Is this in production right now? Do we have the ability to sign up for notifications?

> We're also missing deletion of saved searches

Missed that. Will get an updated version out soon

**Comment by aoduor** [comment]
> Is this in production right now? Do we have the ability to sign up for notifications?

Yup, see here.

![screen capture on 2017-09-27 at 14-33-21](https://user-images.githubusercontent.com/721918/30911338-db801d14-a390-11e7-8f27-87d355873324.gif)

**Comment by jshorland** [comment]
Landing page (map view)
- [ ] name of “timeline” changes to “data” everywhere the name “timeline” occurs

Data view
- [ ] view split into two independently scrolling panes  - left with list of post cards, right with post detail view when a post card is clicked. When no post card is selected, right pane is blank
- [ ] Number of posts based on applied filters displayed above post card list on left pane (4 of 1,203 posts)
- [ ] Checkboxes are removed from post cards and replaced with a right facing chevron after the report titles
- [ ] entire postcard is tappable/clickable for selection except the “…” icon in the top right corner, which brings up the same action sheet as current
- [ ] when selecting a postcard, the postcard nudges to the right —> 0.2s with a quadratic easing curve
- [ ] when a postcard is selected, the post detail appears on the right side pane
- [ ] share button works with current functionality (no change) (so it should share the post detail that’s selected. If no posts are selected, it shares the data view with filtered reports, or exports them if export selected)
Editing a post
- [ ] when postcard is selected, edit button appears next to “share” button if user is logged in with permission to edit posts
- [ ] logged in user with permission to edit posts can navigate directly to edit view through the … button on the postcard (current functionality)
- [ ] When user clicks edit, post detail should enter “edit mode” on the right pane
- [ ] When user clicks cancel without making any changes, taken back to post detail page, no changes saved
- [ ] When user clicks save without making any changes, “you haven’t made any changes error banner” appears. User kept in edit mode
Structuring messages
- [ ] When editing an unstructured post (message via SMS, Twitter or email), user is given a survey dropdown select. Selecting a survey dynamically loads the fields associated with that survey in the right pane
- [ ] user edits as normal once survey is selected. Change log now appears as well 
- [ ] Cancel button - user can cancel without saving changes. 
- [ ] if user navigates away from a post to which changes have been made without saving, Modal appears to either continue without saving or save changes. Continuing without saving changes takes user out of edit mode and back to post detail view of the same post with no changes saved. 
- [ ] clicking the x on the modal, or outside the cancel modal dismisses the modal and keeps user in edit mode
- [ ] Save button - Save changes and automatically select next applicable report





Questions/comments:
- posts or reports??!?! 
- What transition makes the most sense between the post detail view and edit mode?
- We will probably need a loading indicator before the dynamically loaded survey fields while structuring post depending on how long the survey fields take to load. If so, use the “barber stripe” loading indicator on top of the postcard
- do we need to change URL if we change name of timeline to data?

**Comment by jshorland** [comment]
Designs are appended to this post.

**Comment by justinscherer** [comment]
> posts or reports??!?!

Reports, sorry. Trying my best to be consistent with this but having a hard time. 

> What transition makes the most sense between the post detail view and edit mode?

I think the barber strips at the top of the card. The card remains persistent, and it "drops out" with the additional fields when they've loaded. I can prototype this if it proves too tricky. 

>What does "automatically select next applicable report" mean in terms of behaviour?

If you structure something such that it no longer applies to the filters you have enabled, it "swipes" out of your timeline and the next report at the top of the queue becomes selected.

**Comment by justinscherer** [comment]
Quick solve for deleting saved searches.

There's an "edit" link beside the saved searches dropdown in the filter. 
![filter open with filters active](https://user-images.githubusercontent.com/29209303/30955954-64dcbc92-a42d-11e7-9e51-fbc8957bfb7f.png)

It opens a modal with a simple checkbox list that you can save. 
![delete saved searches](https://user-images.githubusercontent.com/29209303/30955978-8019e5e8-a42d-11e7-89f1-8e73f835d23d.png)

**Comment by justinscherer** [comment]
Quick solve for maintaining notifications. You manage notification settings from the "Account settings" modal that comes up when you click your avatar in the bottom left corner. 

I just added another checkbox list to it. 

![notifications](https://user-images.githubusercontent.com/29209303/30956064-cf72a436-a42d-11e7-9c99-e2a781d648fe.png)

**Comment by aoduor** [comment]
My concern with managing notifications solely from my notifications management page is that as a user, I want to be able to quickly identify a saved search that is of interest to me, and get notified. 

* I have to perform 3-4 different clicks to sign up for notifications on a saved search that is of interest to me. 
* In the event that we have a list of many saved searches, it'll take me time to find the one i'm looking for, ( also assumes that I'll automatically remember the name of the specific saved search i want notifications about, in a list of maany)

Makes sense?

**Comment by Angamanga** [comment]
@justinscherer The survey-selector in the post-edit-view, does it appear for all posts or just the ones that still not have a survey selected?

**Comment by justinscherer** [comment]
@aoduor 

> My concern with managing notifications solely from my notifications management page is that as a user, I want to be able to quickly identify a saved search that is of interest to me, and get notified.

There's something to be said for this point, absolutely. However, the current way of doing it is not very discoverable (I didn't even realize it existed until yesterday and I have been working here for months...not that I'm "the User" necessarily). Saved searches are hidden under an unlabeled star icon, then it's still a long list to choose one. Likewise with notifications, it's hidden at the bottom of the global navigation under an avatar and then another weird modal window tab.

If we could get notifications for things other than saved searches we might be in a different spot. Likewise if the information architecture of saved searches wasn't split between the search pane and the big bar on the right of the page. 

At least this way all of the notifications settings are together.

> I have to perform 3-4 different clicks to sign up for notifications on a saved search that is of interest to me.
In the event that we have a list of many saved searches, it'll take me time to find the one i'm looking for, ( also assumes that I'll automatically remember the name of the specific saved search i want notifications about, in a list of maany)
Makes sense?

In either treatment the user has to navigate a massive list of saved searches, especially for uchaguzi. 

Truly, neither the current way of doing things or the quick fixes I just posted are ideal. It will need to get refactored, either way. We need to unravel what categories, tasks, saved searches, and collections actually solve for our users before we can design them in a way that makes sense. 

Right now I'm just trying to plug the holes created by several complicated, entwined features and make the features available in an at least somewhat intuitive way. We / I don't have time/people to do a thorough job, unfortunately. There's a good chance we'll be able to take another stab at this before the end of the year though.

**Comment by justinscherer** [comment]
@Angamanga 

> The survey-selector in the post-edit-view, does it appear for all posts or just the ones that still not have a survey selected?

All posts, even ones that have been structured and published. The idea is that you can change the survey that a report is associated with indefinitely. It's just another field for a report.

**Comment by aoduor** [comment]
@justinscherer 

> If we could get notifications for things other than saved searches we might be in a different spot. Likewise if the information architecture of saved searches wasn't split between the search pane and the big bar on the right of the page.

Currently we get notifications for both collections AND saved searches, so the checkbox list on the notifications box will have both of these.

That said, in total agreement about the need to change this to make this more intuitive, and that we may not have time to do it now.. Just wanted to flag it as a concern we will need to address.

**Comment by Angamanga** [comment]
@justinscherer 
> All posts, even ones that have been structured and published. The idea is that you can change the survey that a report is associated with indefinitely. It's just another field for a report.

We currently do not have this functionality for structured posts so it will need to be implemented. It raises a question we need to address:

If I have a post assigned to survey 1, that has say an image-field, and I want to change to survey 2 that not has an image-field. What do we do with the image? Show a warning? Keep the field while in edit-mode with a clear message somewhere that 'this information will get lost if you proceed to change survey?". Something else? 

I have a memory @willdoran mentioned we had this possibility earlier, @aoduor @jshorland  do you remember if we had and how did we solve it then?

**Comment by justinscherer** [comment]
> If I have a post assigned to survey 1, that has say an image-field, and I want to change to survey 2 that not has an image-field. What do we do with the image? Show a warning? Keep the field while in edit-mode with a clear message somewhere that 'this information will get lost if you proceed to change survey?". Something else?

Good catch. In this case the image would get dropped. We should show a warning that all of the data will be lost when you switch surveys.

**Comment by jshorland** [comment]
@justinscherer Not convinced we need a mockup. We can the loss of data warning screen. @Angamanga is it possible to tell the user WHICH fields of data will be lost on that warning/confirmation screen?

**Comment by Angamanga** [comment]
@justinscherer @jshorland Actually, when I think about it, could we find a way for the user to map each field to new ones?

Just because there are two fields of the same kind in two different surveys, it doesn't not mean they can be translated directly to each other. For example where would the data go if there are two fields of the same kind in the survey you are changing to and only one in the survey you are changing from, or vice versa... Another thing is that the fields might not mean the same thing in two different surveys, for example there might not be the same checkbox-options or different categories in the different surveys 🤔.   Either we just delete everything or make the user point out which old field goes to which new field, and there it would also be visible if some fields would have nowhere to go?

**Comment by willdoran** [comment]
@justinscherer @Angamanga 
>> The survey-selector in the post-edit-view, does it appear for all posts or just the ones that still not have a survey selected?

> All posts, even ones that have been structured and published. The idea is that you can change the survey that a report is associated with indefinitely. It's just another field for a report.

I'm afraid we can't do this in this piece of work, it's a larger element that would require significant design changes and would really be its own feature. For now, we can include the dropdown box to allow structuring of unstructured posts but we won't be able to allow changing of Survey type.

**Comment by justinscherer** [comment]
@willdoran 

>  For now, we can include the dropdown box to allow structuring of unstructured posts but we won't be able to allow changing of Survey type.

So after the user chooses the survey for the first time in the dropdown, we have to disable it against further editing. We'll have to consider it "experience debt" until we can fix it. Where should I log this @jshorland

**Comment by willdoran** [comment]
@justinscherer I just wanna make sure we're talking about the same thing.

As a User, I would be able to change a Report from being Report A to Report B to Report C....to Report N ad infinitum, right? That's going to be super difficult for us. We would need some concept of mapping between Surveys.

In theory, if we have an interface for that it's not hard, we can auto-detect equivalent field types on Survey B and suggest them, the interface would also need to offer the ability to create new fields on the Survey B to import data from Survey A. We would probably also want to provide a means to save that mapping for reuse. 

That isn't an impossible feature but it is a individual involved piece of work in its own right.

**Comment by jshorland** [comment]
We should definitely log a new issue for this. I'm away from my machine at
the moment, but will do it as soon as i land back in seattle

**Comment by justinscherer** [comment]
> As a User, I would be able to change a Report from being Report A to Report B to Report C....to Report N ad infinitum, right? That's going to be super difficult for us. We would need some concept of mapping between Surveys.

Honestly, I just assumed you would reset all of the dynamic fields (i.e. not description or title), and they'd lose any progress there. Probably with a warning. There didn't seem to me to be a need to mapping between fields in multiple surveys. The goal was to alleviate the annoying experience of accidentally putting a report into he wrong survey and not being able to undo it.

**Comment by Angamanga** [comment]
@justinscherer 
> If you structure something such that it no longer applies to the filters you have enabled, it "swipes" out of your timeline and the next report at the top of the queue becomes selected.

Two questions:
1. Should the report selected be in detail or edit-mode?
2. If the filter still applies to the structured post, should that post be the one that is selected?

**Comment by justinscherer** [comment]
> Should the report selected be in detail or edit-mode?

Detail mode. You have e to click edit in the top right corner to enable edit mode. 

> If the filter still applies to the structured post, should that post be the one that is selected?

Yes.

**Comment by Angamanga** [comment]
@justinscherer @middle8media Realising we need to look at the behaviour of the layout-panes on smaller screens too. As it is now, the post-details/post-edit disappear when on smaller screens than ~1000 px. What should the behaviour be there?

**Comment by justinscherer** [comment]
@Angamanga 

On smaller screens, postdetails/postedit will open into a new view like the current behaviour.

**Comment by Angamanga** [comment]
Things to remember:
- [ ] when I navigate to data from a collection, it should go to data-view
- [ ] when I navigate to a saved search after saving it, it should default to where?
